subworkflows/local/bgc.nf

```diff
@@ -0,0 +1,106 @@
+    // Important for future maintenance: For CI tests, only the first of the 4 options below is used. Thus, all 4 combinations below should be tested locally whenever the antiSMASH module gets updated.
+    if ( params.bgc_antismash_databases && params.bgc_antismash_directory ) {
+
+        // Supply databases
+        ch_antismash_databases = Channel
+            .fromPath( params.bgc_antismash_databases )
+            .first()
+
+        // Supply antiSMASH directory
+        ch_antismash_directory = Channel
+            .fromPath( params.bgc_antismash_directory )
+            .first()
+
+    } else if ( params.bgc_antismash_databases && !params.bgc_antismash_directory ) {
+
+        // Supply databases
+        ch_antismash_databases = Channel
+            .fromPath( params.bgc_antismash_databases )
+            .first()
+
+        // Supply antiSMASH directory
+        ch_css_for_antismash = "https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/antismash/css.tar.gz"
+        ch_detection_for_antismash = "https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/antismash/detection.tar.gz"
+        ch_modules_for_antismash = "https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/antismash/modules.tar.gz"
+
+        UNTAR_CSS ( [ [], ch_css_for_antismash ] )
+        UNTAR_DETECTION ( [ [], ch_detection_for_antismash ] )
+        UNTAR_MODULES ( [ [], ch_modules_for_antismash ] )
+
+        ANTISMASH_ANTISMASHLITEDOWNLOADDATABASES ( UNTAR_CSS.out.untar.map{ it[1] }, UNTAR_DETECTION.out.untar.map{ it[1] }, UNTAR_MODULES.out.untar.map{ it[1] } )
+        ch_versions = ch_versions.mix(ANTISMASH_ANTISMASHLITEDOWNLOADDATABASES.out.versions)
+        ch_antismash_directory = ANTISMASH_ANTISMASHLITEDOWNLOADDATABASES.out.antismash_dir
+
+    } else if ( !params.bgc_antismash_databases && params.bgc_antismash_directory ) {
+
+        // Supply databases
+        ch_css_for_antismash = "https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/antismash/css.tar.gz"
+        ch_detection_for_antismash = "https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/antismash/detection.tar.gz"
+        ch_modules_for_antismash = "https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/antismash/modules.tar.gz"
+
+        UNTAR_CSS ( [ [], ch_css_for_antismash ] )
+        UNTAR_DETECTION ( [ [], ch_detection_for_antismash ] )
+        UNTAR_MODULES ( [ [], ch_modules_for_antismash ] )
+
+        ANTISMASH_ANTISMASHLITEDOWNLOADDATABASES ( UNTAR_CSS.out.untar.map{ it[1] }, UNTAR_DETECTION.out.untar.map{ it[1] }, UNTAR_MODULES.out.untar.map{ it[1] } )
+        ch_versions = ch_versions.mix(ANTISMASH_ANTISMASHLITEDOWNLOADDATABASES.out.versions)
+        ch_antismash_databases = ANTISMASH_ANTISMASHLITEDOWNLOADDATABASES.out.database
+
+        // Supply antiSMASH directory
+        ch_antismash_directory = Channel
+            .fromPath( params.bgc_antismash_directory )
+            .first()
+
```

Comment:
Just for my understanding: What would be the reason for a user to supply either only directory or only database? In these two cases ANTISMASH_ANTISMASHLITEDOWNLOADDATABASES has to be run anyway, with the only difference that one of the outputs will not be used, right? 
Wouldn't it make sense to restrict it to either supply both parameters or none? (like if only one is given print a reminder to supply both?). It would make the code less repetitive.

Reply:
>What would be the reason for a user to supply either only directory or only database? In these two cases ANTISMASH_ANTISMASHLITEDOWNLOADDATABASES has to be run anyway, with the only difference that one of the outputs will not be used, right?

Exactly!

>Wouldn't it make sense to restrict it to either supply both parameters or none? (like if only one is given print a reminder to supply both?). It would make the code less repetitive.

It _can_ happen that you only have a local database folder but not a installation directory, but is not very likely. Clean code is always desirable, I'll adapt your suggestion :)